app/utils/runtime-environment.ts

```diff
@@ -0,0 +1,10 @@
+declare const RUNTIME_ENVIRONMENT: Record<string, string>;
+
+// NOTE: This is where we figure out if we use the client-side or server-side runtime environment
+export default (key: string) => {
+  if (typeof window !== 'undefined' && window.RUNTIME_ENVIRONMENT) {
```

Comment:
Why do we have to explicitly check the type of `window` instead of using _optional chaining_?

```suggestion
  if (window?.RUNTIME_ENVIRONMENT) {
```